### PR TITLE
Migrate certificate template to mmv1

### PR DESCRIPTION
**Pull request by trodge**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:note
privateca: `google_privateca_certificate_template` now uses mmv1 engine instead of DCL
```

### CODE REVIEW

File: mmv1/products/privateca/CertificateTemplate.yaml [hunk 1]
```diff
@@ -12,14 +12,36 @@
...
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
 description: |
-  Only used to generate IAM resources
+  A CertificateTemplate refers to a managed template for certificate issuance.
```
Comment (slevenick) [inline mmv1/products/privateca/CertificateTemplate.yaml:43]: Looks like this originally had a more in-depth description, can you add that back?
https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10527-old..auto-pr-10527#diff-8fc7d1e5f797d25b4b2bdb42ae8679906db4a7b151a3ec25adf939ef65aaa098L18

File: mmv1/products/privateca/CertificateTemplate.yaml [hunk 2]
```diff
@@ -12,14 +12,36 @@
...
+create_url: projects/{{project}}/locations/{{location}}/certificateTemplates?certificateTemplateId={{name}}
 self_link: projects/{{project}}/locations/{{location}}/certificateTemplates/{{name}}
-# This resource is only used to generate IAM resources. They do not correspond to real
-# GCP resources, and should not be used to generate anything other than IAM support.
-exclude_resource: true
+update_verb: :PATCH
+update_mask: true
+references: !ruby/object:Api::Resource::ReferenceLinks
+  guides:
+    'Official Documentation': 'https://cloud.google.com/certificate-authority-service'
```
Comment (slevenick) [inline mmv1/products/privateca/CertificateTemplate.yaml:23]: Looks like the existing resource has additional guides linked: https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10527-old..auto-pr-10527#diff-8fc7d1e5f797d25b4b2bdb42ae8679906db4a7b151a3ec25adf939ef65aaa098L26

File: mmv1/products/privateca/CertificateTemplate.yaml [hunk 3]
```diff
@@ -35,8 +57,241 @@ examples:
     primary_resource_id: 'default'
     vars:
       name: 'my-template'
+parameters:
+  - !ruby/object:Api::Type::String
+    name: location
```
Comment (slevenick) [inline mmv1/products/privateca/CertificateTemplate.yaml:62]: This seems to have changed the behavior (or at least documentation) of the IAM resource based on this.

Do you know what is going on there?

https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-10527-old..auto-pr-10527#diff-5dcfd60d59ef2ac2f119d71bdf3847bfc09604afa7e388e8abc050ebcc489bf0R140

**Review by slevenick — COMMENTED** [review]
Can you add the example that was used in the docs generation for the DCL resource so that we don't have a big diff in the docs page?

File: mmv1/products/privateca/CertificateTemplate.yaml (hunk 3 above)
Comment (trodge) [reply mmv1/products/privateca/CertificateTemplate.yaml:62]: It looks like there's a bug for all mmv1 resources where the `location` field is documented as required on iam policy even though it is actually optional. I have fixed that in this PR.

**Pull request merged**